tools/cli-bootstrap/README.md

```diff
@@ -0,0 +1,36 @@
+# Rover
+The dotnet CLI bootstrapping tool
+
+## What
+Bootstrapping a tool like the dotnet CLI can be a dizzying effort. From a high-level we simply want to replace the native components of a pre-existing CLI with new ones that
+have been built on the targetted platform. Unfortunately, from the bottom-up, this means that we need to clone a handful of dotnet repositories, execute particular
+command lines with particular arguments and then copy them in to a pre-existing dotnet file. Rover aims to simplify that.
+
+The goal is to place Rover on to the distro of choice, execute it, and then have it do the required steps for you - and then things 'just work'! 
+
+However, things don't always 'just work.' When this is the case Rover hopes to reduce the amount of effort necessary to fix build or script breaks by placing 'repro' shell scripts when there is a failure so that a developer can go in and 'apply pressure' where appropriate. 
+
+## How
+
+There are default settings for each of the rover arguments. If all is going to plan, you should never need to specify any additional parameters. However
+many situations arise where it becomes a necessity or a nice-to-have. Namely, when things go wrong, you want to be able to get in there and fix
+up the build, make changes, etc. This is 'development mode' or DevMode for short. In DevMode, NO git commands are executed. This is to prevent 
+the script from stomping out any changes you have made in the working directory. 
+
+DevMode is triggered if the working directory is pre-existing. Consequently, when things fail in the script, we do not 'clean up' the working directory. We write a repro script
```

Comment:
Ah yes. I will revise the docs a bit overall. There is a bunch of stuff here that I do not like or is even a bit out-dated already.